Modules/IO/ImageBase/include/itkImageFileReader.h

```diff
@@ -167,6 +167,21 @@ class ITK_TEMPLATE_EXPORT ImageFileReader : public ImageSource<TOutputImage>
   // produce the requested region.
   ImageIORegion m_ActualIORegion;
 };
+
+
+/** Convenience function for reading an image. */
```

Comment:
This description seems short.

A description should include that the function updated the whole image. Also perhaps a description of the template parameters or the behaviors of the pixel trains template parameter would be good.

Reply:
> A description should include that the function updated the whole image.

Thanks again @blowekamp but when a user does `auto image = itk::ReadImage(fileName)`, it seems obvious to me that the whole image is updated. A user cannot do something like `image->Update()` afterwards anyway. OK?

I do agree with you that a description of the template parameter `ConvertPixelTraits` would be good. I'll add one with the next force-push. 😃

Reply:
I may be wrong, but I thought one of the ITK python procedural methods return objects ( Image or Filter?) that still needed to be updated?

I think executing `image->Update()` afterwords would still work. While its not this case, in others, this would have the side effect of setting the LargestPossibleRegion to the BufferedRegion.

Reply:
OK, I did not even know that `image->Update()` would work. But then again, isn't it obvious to you that the image returned by `ReadImage` is the image that was just read?

Reply:
It is obvious. The docstring is for people who are having some doubts, maybe because they know streaming exists in ITK 😄 and want to check whether this convenience function participates.

Reply:
@blowekamp @dzenanz Just extended the function description, hope you appreciate 😃 I added the following sentence:

> The function reads the image from the specified file, and returns the image that it has read.

https://github.com/InsightSoftwareConsortium/ITK/blob/3778f98e1f4d2edc53c135d8526ed2ab3da665d5/Modules/IO/ImageBase/include/itkImageFileReader.h#L178-L179